Updated Vuex Section Documentation

capndave opened the pull request:
Hi I added a section to the docs on how to emit events from the Vuex store. I edited the entire section for grammar and removed/simplified some of the language that I found confusing or repetitive. Hopefully you'll find it clearer and easier to read! Please take or leave whichever parts you find helpful. I'm happy to work more on the docs or to tackle other issues if you like as well:)

probil commented on README.md:
````diff
@@ -209,11 +210,21 @@ export default new Vuex.Store({
+```
+  actions: {
+    emitSocketEvent (data) {
+      this._vm.$socket.emit('eventName', JSON.stringify(data))     
````
The API was changed in v4:
```diff
- this._vm.$socket.emit
+ this._vm.$socket.client.emit
```

[Check release docs](https://github.com/probil/vue-socket.io-extended/releases/tag/v4.0.0)

capndave commented:
I changed `$socket.emit` to  `$socket.client.emit`, and added syntax highlighting. Unfortunately you can't do inline code with syntax highlighting in markdown, so it doesn't read quite as nicely. I'll change it back to inline w/o syntax highlighting if you prefer, but here it is!

probil commented on README.md:
````diff
@@ -209,11 +210,25 @@ export default new Vuex.Store({
+
+Events can be sent to the Socket.IO server by calling 
+```js
+this._vm.$socket.client.emit
````
Use it like so: `this._vm.$socket.client.emit` 
no need for new lines and syntax highlight here

I'll merge after that change

capndave replied:
Awesome, thanks!

probil commented on README.md:
````diff
@@ -209,11 +210,25 @@ export default new Vuex.Store({
+```js
+this._vm.$socket.client.emit
+```
+from a Vuex mutation or action. Mutation or action names are not subject to the same naming requirements as above. Event data must be sent in string format.
````
Event data can be send in any format (not just string)
https://socket.io/docs/client-api/#socket-emit-eventName-%E2%80%A6args-ack

capndave replied:
Good to know, thanks!

probil commented:
@capndave merged as part of #414
